packages/actor-query-source-identify-hypermedia/lib/MediatedLinkedRdfSourcesAsyncRdfIterator.ts

```diff
@@ -144,9 +153,22 @@ export class MediatedLinkedRdfSourcesAsyncRdfIterator extends LinkedRdfSourcesAs
     return this.linkQueue;
   }
 
-  protected async getSourceLinks(metadata: Record<string, any>): Promise<ILink[]> {
+  protected async getSourceLinks(metadata: Record<string, any>, startSource: ISourceState): Promise<ILink[]> {
```

Comment:
Is keeping track of this `startSource` really needed, given that we have `this.firstUrl`?

Reply:
Yes, as `this.firstUrl` will always refer to the first URL that is dereferenced and not the link responsible (the parent of the found links in this case) for the call to getSourceLinks.

Reply:
I believe it's because during link traversal `MediatedLinkedRdfSourcesAsyncRdfIterator` is only made once. The class attribute `this.firstUrl` is then set to the seed document. During traversal a URL in the queue is dereferenced [here](https://github.com/comunica/comunica/blob/721bdf88a407e769e57a25102df6151f1021b0f0/packages/actor-query-source-identify-hypermedia/lib/LinkedRdfSourcesAsyncRdfIterator.ts#L301). From this URL, new URLs are discovered [here](https://github.com/comunica/comunica/blob/721bdf88a407e769e57a25102df6151f1021b0f0/packages/actor-query-source-identify-hypermedia/lib/LinkedRdfSourcesAsyncRdfIterator.ts#L256) and the next iterator in the queue is started [here](https://github.com/comunica/comunica/blob/721bdf88a407e769e57a25102df6151f1021b0f0/packages/actor-query-source-identify-hypermedia/lib/LinkedRdfSourcesAsyncRdfIterator.ts#L270).  In this process, no new `MediatedLinkedRdfSourcesAsyncRdfIterator` is made, so `this.firstUrl` does not (most of the time) correspond to the parent URL of newly discovered URLs.

Reply:
Ah nvm, I misread the diff.
I thought this change was in `getSourceLinks`, but it's in `_read`.

Looks good in that case!